libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -1511,45 +1522,56 @@ void EthStratumClient::sendSocketData(Json::Value const& jReq)
     if (!isConnected())
         return;
 
-    // Out received message only for debug purpouses
-    if (g_logOptions & LOG_JSON)
-    {
-        cnote << jReq;
-    }
-
-    std::ostream os(&m_sendBuffer);
-    os << m_jWriter.write(jReq);  // Do not add lf. It's added by writer.
-
-    if (m_conn->SecLevel() != SecureLevel::NONE)
-    {
-        async_write(*m_securesocket, m_sendBuffer,
-            m_io_strand.wrap(boost::bind(&EthStratumClient::onSendSocketDataCompleted, this,
-                boost::asio::placeholders::error)));
-    }
-    else
-    {
-        async_write(*m_nonsecuresocket, m_sendBuffer,
-            m_io_strand.wrap(boost::bind(&EthStratumClient::onSendSocketDataCompleted, this,
-                boost::asio::placeholders::error)));
-    }
+    // Need a copy of the value for the Q, parameter is ephemerous
+    Json::Value *j = new Json::Value(jReq);
```

Comment:
Use a queue of string and push the message already as string. 
Here instead you have a Json object which copy creates another json object which will be eventually transformed to string. One pass can be removed.